src/main/java/de/adesso/kicker/notification/NotificationService.java

```diff
@@ -0,0 +1,104 @@
+package de.adesso.kicker.notification;
+
+import de.adesso.kicker.notification.MatchVerificationRequest.MatchVerificationRequest;
+import de.adesso.kicker.notification.MatchVerificationRequest.VerifyMatchService;
+import de.adesso.kicker.notification.exception.NotificationNotExistingException;
+import de.adesso.kicker.notification.exception.WrongReceiverException;
+import de.adesso.kicker.user.User;
+import de.adesso.kicker.user.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.text.MessageFormat;
+import java.util.List;
+import java.util.Properties;
+
+@Service
+public class NotificationService {
+
+    private NotificationRepository notificationRepository;
+    private VerifyMatchService verifyMatchService;
+    private UserService userService;
+
+    @Autowired
+    public NotificationService(NotificationRepository notificationRepository, VerifyMatchService verifyMatchService,
+            UserService userService) {
+
+        this.notificationRepository = notificationRepository;
+        this.verifyMatchService = verifyMatchService;
+        this.userService = userService;
+    }
+
+    public void acceptNotification(long notificationId) {
+
+        Notification notification = notificationRepository.findByNotificationId(notificationId);
+        checkNotificationExists(notification);
+        checkWrongReceiver(notification);
+        switch (notification.getType()) {
+        case MESSAGE:
+            deleteNotification(notification);
+            break;
+        case MATCH_VERIFICATION:
+            verifyMatchService.acceptRequest((MatchVerificationRequest) notification);
+            break;
+        }
+    }
+
+    public void declineNotification(long notificationId) {
+
+        Notification notification = notificationRepository.findByNotificationId(notificationId);
+        checkNotificationExists(notification);
+        checkWrongReceiver(notification);
+        switch (notification.getType()) {
+        case MESSAGE:
+            deleteNotification(notification);
+            break;
+        case MATCH_VERIFICATION:
+            // TODO delete the match
+            List<User> users = verifyMatchService.declineRequest((MatchVerificationRequest) notification);
+            Properties properties = new Properties();
+            String message = MessageFormat.format(properties.getProperty("notification.message.declined"),
+                    userService.getLoggedInUser().getFirstName(),
+                    ((MatchVerificationRequest) notification).getMatch().getDate().toString());
+            for (User user : users) {
+                sendNotification(null, user, message);
```

Comment:
Here you create a `MESSAGE` which doesn't require  a sender. Would it be possible to remove `sender` from `Notification` and add it to `MatchVerificationRequest`, as a `MESSAGE` doesn't need a sender apparently?

Reply:
Yes it would be possible.
But if some new Notification types are goind to b implemented in the future which may also have an sender we could have this attribute generalized directly.

Reply:
Alright I see what you mean. Theoretically we could add another layer of inheritance one for notifications with and one without sender but I guess that would be a bit overkill.